src/applications/train/run_mlm_pytorch_stream.py

```diff
@@ -0,0 +1,775 @@
+"""
+Fine-tuning the huggingface models for masked language modeling (BERT, ALBERT,
+RoBERTa...) on a text file or a dataset.
+
+Outline copied from:
+https://github.com/huggingface/transformers/blob/main/examples/pytorch/language-modeling/run_mlm.py
+Adapted for streaming datasets.
+
+
+Here is the full list of checkpoints on the hub that can be fine-tuned by this script:
+https://huggingface.co/models?filter=fill-mask
+
+
+Note: requires the dev version of HF datasets:
+https://github.com/huggingface/datasets
+
+Example usage:
+
+.. code::
+
+    python src/applications/train/run_mlm_pytorch_stream.py \
+        --train_file test.txt \
+        --model_name_or_path roberta-base \
+        --output_dir /tmp/test_mlm \
+        --do_train \
+        --overwrite_output_dir \
+        --streaming \
+        --validation_split 1000 \
+        --max_steps 1 \
+        --max_train_samples 1000
+
+"""
+
+import logging
+import math
+import os
+import sys
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import Optional
+
+import datasets
+import transformers
+from datasets import load_dataset, load_metric
+from transformers import (
+    CONFIG_MAPPING,
+    MODEL_FOR_MASKED_LM_MAPPING,
+    AutoConfig,
+    AutoModelForMaskedLM,
+    AutoTokenizer,
+    DataCollatorForLanguageModeling,
+    HfArgumentParser,
+    Trainer,
+    TrainingArguments,
+    is_torch_tpu_available,
+    set_seed,
+)
+from transformers.trainer_utils import get_last_checkpoint
+
+logger = logging.getLogger(__name__)
+MODEL_CONFIG_CLASSES = list(MODEL_FOR_MASKED_LM_MAPPING.keys())
+MODEL_TYPES = tuple(conf.model_type for conf in MODEL_CONFIG_CLASSES)
+
+
+@dataclass
+class ModelArguments:
+    """
+    Arguments pertaining to which model/config/tokenizer we are going to fine-tune, or train from scratch.
+    """
```

Comment:
I think docstring readers use the first line as the description.

```suggestion
    """Arguments pertaining to which model/config/tokenizer we are going to fine-tune, or train from scratch."""
```